packages/cli/src/commands/export-dynamic-plugin/dev.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PackageRoleInfo } from '@backstage/cli-node';
+import { ConfigReader } from '@backstage/config';
+import { loadConfig } from '@backstage/config-loader';
+
+import chalk from 'chalk';
+import { OptionValues } from 'commander';
+import * as fs from 'fs-extra';
+
+import path from 'path';
+
+import { paths } from '../../lib/paths';
+import { Task } from '../../lib/tasks';
+
+export async function applyDevOptions(
+  opts: OptionValues,
+  pkgName: string,
+  role: PackageRoleInfo,
+  target: string,
+) {
+  if (opts.dev) {
+    if (role.platform === 'node') {
+      await fs.ensureSymlink(
+        paths.resolveTarget('src'),
+        path.resolve(target, 'src'),
+        'dir',
+      );
+    }
+
+    let dynamicPluginsRootPath = opts.dynamicPluginsRoot as string | undefined;
+    let shouldSymlink = false;
+    if (!dynamicPluginsRootPath) {
+      shouldSymlink = true;
```

Comment:
@davidfestal I feel like you probably mentioned this to me during the call, but why not allow symlinking when using `--dynamic-plugins-root`?

Reply:
Because it might be outside of the current backstage monorepo, in which case (due to how the NodeJS module loader resolves symlinks), the dynamically-loaded plugins will not find the shared dependencies of the backstage instance you want to use.